main.go

```diff
@@ -326,10 +342,19 @@ func main() {
 	log.Printf("Took: " + time.Since(startTime).String())
 }
 
-func isSameStack(archiveStackID string, currentStackID string) bool {
+func isSameStack(archiveStackInfo archiveInfo, currentStackInfo archiveInfo) bool {
 	// TODO This check is a temporary solution to support GEN2 VMs having different ids for same stack types
 	r := regexp.MustCompile("^(.+)-gen2.*$")
-	currentStackID = r.ReplaceAllString(currentStackID, "$1")
-	archiveStackID = r.ReplaceAllString(archiveStackID, "$1")
-	return archiveStackID == currentStackID
+	currentStackInfo.StackID = r.ReplaceAllString(currentStackInfo.StackID, "$1")
+	archiveStackInfo.StackID = r.ReplaceAllString(archiveStackInfo.StackID, "$1")
+
+	if archiveStackInfo.StackID != currentStackInfo.StackID {
+		return false
+	}
+
+	if archiveStackInfo.Version < 2 {
```

Comment:
Introduced the version 2 as an arbitrary number (it did not exists before).

Reply:
You mentioned backwards compatibility but do we need it or can we guarantee that it will work? If I remember correctly (this might not be accurate) but some customers already have access to M1 machines to test them. So the cache archives without this new version field could be already generated on two different machine types. 

I would just not use the cache if we cannot determine the machine type it was generated on.

Reply:
Removed backwards compatibility as per discussion.